src/main/java/io/github/christophermanahan/carnitas/Router.java

```diff
@@ -45,12 +58,26 @@ private HTTPResponse handleRoute(HTTPRequest request) {
         ).apply(request);
     }
 
-    private boolean routeAdded(HTTPRequest request) {
+    private boolean allowedUri(HTTPRequest request) {
         return map.keySet().stream()
-          .anyMatch(route -> route.equals(route(request)));
+          .anyMatch(route -> route.uri().equals(request.uri()));
     }
 
-    private Route route(HTTPRequest request) {
-        return new Route(request.method(), request.uri());
+    private HTTPResponse methodNotAllowed(HTTPRequest request) {
+        return new HTTPResponse(HTTPResponse.Status.METHOD_NOT_ALLOWED)
```

Comment:
I found this hard to read. I'm wondering if you can extract parts of it into variables names to be more descriptive

Reply:
I agree, this is difficult to read and I believe this ties into your other comment about the Router having multiple responsibilities. Separating those concerns may alleviate this syntax.

Reply:
I have to keep in mind that my IPM is at 2 today and I would like to get this PR merged so that I can demo this feature this afternoon. I do not think it would be wise to take on a refactor of HTTPResponse with so little time left. I will take on some technical debt that will be addressed in the next iteration.